### ISSUE TITLE
Look up code unit by filename and line number

### ISSUE THREAD

**Issue by sebastianbergmann** [Roave/BetterReflection#178]
I recently needed functionality to [look up a code unit by filename and line number](https://github.com/sebastianbergmann/code-unit-reverse-lookup/blob/master/src/Wizard.php#L39) and put it into a new component that is now a dependency of php-code-coverage.

I am not happy with the names that I have chosen for this functionality and begin to think that this functionality may be better kept in a component with similar functionality. I was wondering if you already have this in BetterReflection or would consider adding it if not?

**Comment by asgrim** [comment]
Yes; this can be done quite trivially; there's a couple of ways I can think of doing this but I need to have a think through them to make sure it's exactly what you're after (whether we need to change/add functionality, or whether it can be done already with minimal fuss for example)

**Comment by sebastianbergmann** [comment]
Thanks for looking into it, @asgrim. Let me know if there is anything I can help with.

**Comment by asgrim** [comment]
Alright, I've had a look, and we can pretty easily add new methods to our `ClassReflector` and `FunctionReflector` to `reflectOnLine`. So the usage would look something like this:

```
function findClass($file, $line)
{
    return (new ClassReflector(new SingleFileSourceLocator($file)))->reflectOnLine($line);
}
function findFunct($file, $line)
{
    return (new FunctionReflector(new SingleFileSourceLocator($file)))->reflectOnLine($line);
}
```

This would work for interfaces, classes, traits and functions. You could combine those two calls to get "anything" on a line.

If you're happy with this sort of usage, I can finish the implementation and get this merged, we'd target the 1.1.0 release for this.

**Comment by sebastianbergmann** [comment]
It would be great if there was a `findAll($file, $line)` method as it would be inconvenient to first try `findClass()` and then `findFunct()` if the first call did not yield anything.

**Comment by asgrim** [comment]
@sebastianbergmann feedback welcome on PR #179 - from what I understand this fulfils your needs now?

> As per discussion in feature request #178, this introduces a new `CombinedReflector` API that allows reflecting on a specific line.
> 
> Usage would look like this:
> 
> ```php
> $reflector = new CombinedReflector(new SingleFileSourceLocator($file));
> 
> // Will return a ReflectionFunction, ReflectionClass or null, depending on
> // what is on line 7 in $file
> $item = $reflector->reflectOnLine(7);
> ```

**Comment by asgrim** [comment]
@sebastianbergmann I've had a rethink and made a new PR #181, which should encapsulate your use case perfectly, usage is also even simpler (by way of an invokable class):

```php
<?php

$finder = new FindReflectionOnLine();
$reflection = $finder('path/to/my/file.php', 10);
```

**Comment by Ocramius** [comment]
Feature was provided via #181

**Comment by asgrim** [comment]
@sebastianbergmann FYI, this feature is now released in our latest stable version (1.1.0) :tada:
